shell/profiling/profiler.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/profiling/profiler.h"
+
+namespace flutter {
+
+Profiler::Profiler(fml::RefPtr<fml::TaskRunner> profiler_task_runner,
+                   CpuUsage cpu_usage,
+                   int num_samples_per_sec)
+    : profiler_task_runner_(profiler_task_runner),
+      cpu_usage_(std::move(cpu_usage)),
+      num_samples_per_sec_(num_samples_per_sec) {}
+
+void Profiler::Start() {
+  if (!profiler_task_runner_) {
+    return;
+  }
+  FML_CHECK(num_samples_per_sec_ > 0)
+      << "number of samples must be a positive integer, got: "
+      << num_samples_per_sec_;
+  double delay_between_samples = 1.0 / num_samples_per_sec_;
+  auto task_delay = fml::TimeDelta::FromSecondsF(delay_between_samples);
+  RunTask(task_delay);
+}
+
+void Profiler::RunTask(fml::TimeDelta task_delay) {
```

Comment:
Nit: rename `RunTask(task_delay)` to `SampleRepeatedly(sample_interval)`? `RunTask` seems to suggest that it's running only once.

Reply:
Done.